### ISSUE TITLE
Failed derivative mutates the mathematical expression

### ISSUE THREAD

**Issue by sylee957** [josdejong/mathjs#3094]
**Describe the bug**

If I try-catch the evaluation of derivative, the expression gets unexpected mutation in-place if it fails 
I'm using `12.0.0`

**To Reproduce**
Steps to reproduce the behavior.

https://codesandbox.io/s/fast-worker-pnq7lf?file=/src/App.tsx

```
const expr = math.parse("min(x, y)");
console.log(math.format(expr));
try {
  const dx = math.derivative(expr, "x");
} catch {}
console.log(math.format(expr));
```
```
min(x, y) 
min(0, 0)
```

**Comment by josdejong** [comment]
Thanks for reporting! 

Fixed via a1f3b7c6bf8e748d56c93303844e89e4f3d8ce37 (not yet published)

**Comment by josdejong** [comment]
The fix is published now in `v12.1.0`
